neuralprophet/configure.py

```diff
@@ -140,6 +126,39 @@ def __post_init__(self):
         if len(self.quantiles) > 1:
             self.loss_func = PinballLoss(loss_func=self.loss_func, quantiles=self.quantiles)
 
+    def set_quantiles(self):
+        # assert either prediction interval or quantiles is None, or both are None
+        assert self.prediction_interval is None or self.quantiles is None, "Prediction interval and quantiles cannot both be populated, one or both must be None."
+        if self.uncertainty_method.lower() in ["quantile_regression", "quantile regression", "qr"]:
+            # assert prediction interval is None and quantiles is a list
+            assert self.prediction_interval is None and isinstance(self.quantiles, list), "Prediction interval must be None while quantiles must be a list. Otherwise , set uncertainty estimation param to 'auto'"
+        elif self.uncertainty_method.lower() in ["conformal_prediction", "conformal prediction", "cp"]:
+            # assert prediction interval is a float and quantiles is None
+            assert isinstance(self.prediction_interval, float) and self.quantiles is None, "Prediction interval must be a float while quantiles must be None. Otherwise , set uncertainty estimation param to 'auto'"
+        elif self.uncertainty_method.lower() in ["auto", "a"]:
+            # assert that prediction interval is a float between (0, 1) if not None, then use that to create the quantiles
+            if self.prediction_interval is not None:
```

Comment:
If quantiles are provided, this would overwrite them. I think quantiles are more explicit, and thus they should take precedence.
I suggest to change this line to:
`if self.prediction_interval is not None and self.quantiles is None:`
-> also, after setting quantiles, set `self.prediction_interval = None`

and additionally
`elif self.prediction_interval is not None and self.quantiles is None:`
-> log.warn("Ignoring prediction_interval and using specified quantiles for uncertainty estimation")
and also: set `self.prediction_interval = None`

Reply:
The first assertion statement prevents both `self.prediction_interval` and `self.quantiles` from beings values other than `None`.

`assert self.prediction_interval is None or self.quantiles is None, "Prediction interval and quantiles cannot both be populated, one or both must be None."`

Nevertheless, for clarity, I'll still change this line to `if self.prediction_interval is not None and self.quantiles is None:` and set `self.prediction_interval = None` after.

Reply:
I pushed, which incorporated your suggestions for the first half. However, I'm still unsure about what you meant regarding the part after "additionally".

Reply:
the part of after _additionally_ was refering to handling the case when we would not raise an error if both `prediction_interval` and `quantiles` are populated.

Reply:
But we can raise an error in this case as you configured now, no need to change.